studio/components/interfaces/Settings/Integrations/IntegrationsSettings.tsx

```diff
@@ -0,0 +1,830 @@
+import { useBranchQuery } from 'data/branches/branch-query'
+import { useBranchesQuery } from 'data/branches/branches-query'
+import { GitBranch, Shield, RotateCcw, GitCompare } from 'lucide-react'
+import { useVercelConnectionUpdateMutation } from 'data/integrations/vercel-connection-update-mutate'
+import Link from 'next/link'
+
+const IntegrationImageHandler = ({ title }: { title: 'vercel' | 'github' }) => {
+  return (
+    <img
+      className="border rounded-lg shadow w-48 mt-6 border-body"
+      src={`${BASE_PATH}/img/integrations/covers/${title}-cover.png`}
+      alt={`${title} cover`}
+    />
+  )
+}
+
+const IntegrationSettings = () => {
+  const org = useSelectedOrganization()
+  const { data } = useOrgIntegrationsQuery({ orgSlug: org?.slug })
+  const sidePanelsStateSnapshot = useSidePanelsStateSnapshot()
+
+  const projectContext = useProjectContext()
+
+  const vercelIntegrations = useMemo(() => {
+    return data
+      ?.filter((integration) => integration.integration.name === 'Vercel')
+      .map((integration) => {
+        if (integration.metadata && integration.integration.name === 'Vercel') {
+          const avatarSrc =
+            !integration.metadata.account.avatar && integration.metadata.account.type === 'Team'
+              ? `https://vercel.com/api/www/avatar?teamId=${integration.metadata.account.team_id}&s=48`
+              : `https://vercel.com/api/www/avatar/${integration.metadata.account.avatar}?s=48`
+
+          integration['metadata']['account']['avatar'] = avatarSrc
+        }
+
+        return integration
+      })
+  }, [data])
+
+  const githubIntegrations = data?.filter(
+    (integration) => integration.integration.name === 'GitHub'
+  )
+
+  // We're only supporting one Vercel integration per org for now
+  // this will need to be updated when we support multiple integrations
+  const vercelIntegration = vercelIntegrations?.[0]
+  const { data: vercelProjectsData } = useVercelProjectsQuery(
+    {
+      organization_integration_id: vercelIntegration?.id,
+    },
+    { enabled: vercelIntegration?.id !== undefined }
+  )
+  const vercelProjectCount = vercelProjectsData?.length ?? 0
+
+  const onAddVercelConnection = useCallback(
+    (integrationId: string) => {
+      sidePanelsStateSnapshot.setVercelConnectionsIntegrationId(integrationId)
+      sidePanelsStateSnapshot.setVercelConnectionsOpen(true)
+    },
+    [sidePanelsStateSnapshot]
+  )
+
+  const onAddGitHubConnection = useCallback(
+    (integrationId: string) => {
+      sidePanelsStateSnapshot.setGithubConnectionsIntegrationId(integrationId)
+      sidePanelsStateSnapshot.setGithubConnectionsOpen(true)
+    },
+    [sidePanelsStateSnapshot]
+  )
+
+  const { mutateAsync: deleteVercelConnection } =
+    useIntegrationsVercelInstalledConnectionDeleteMutation()
+
+  const onDeleteVercelConnection = useCallback(
+    async (connection: IntegrationProjectConnection) => {
+      await deleteVercelConnection({
```

Comment:
needs to be in a try catch cause its mutateAsync, unless we do mutate instead then we chuck onSuccess and onError handlers when calling useIntegrationsVercelInstalledConnectionDeleteMutation()